src/client/QXmppPubSubManager.cpp

```diff
@@ -305,6 +344,35 @@ auto QXmppPubSubManager::deleteNode(const QString &jid, const QString &nodeName)
     return client()->sendGenericIq(std::move(request));
 }
 
+///
+/// Requests the IDs of all items of a pubsub service node via service
+/// discovery.
+///
+/// This uses a \xep{0030, Service Discovery} items request to get a list of
+/// items.
+///
+/// \param serviceJid JID of the entity hosting the pubsub service
+/// \param nodeName the name of the node whose items are requested
+/// \return
+///
+QFuture<QXmppPubSubManager::ItemIdsResult> QXmppPubSubManager::requestItemIds(const QString &serviceJid, const QString &nodeName)
+{
+    QXmppDiscoveryIq request;
+    request.setType(QXmppIq::Get);
+    request.setQueryType(QXmppDiscoveryIq::ItemsQuery);
+    request.setQueryNode(nodeName);
+    request.setTo(serviceJid);
+
+    return chainIq(client()->sendIq(std::move(request)), this, [](QXmppDiscoveryIq &&iq) -> ItemIdsResult {
+        const auto queryItems = iq.items();
+        QVector<QString> itemIds;
+        for (const auto &queryItem : queryItems) {
+            itemIds << queryItem.name();
+        }
+        return itemIds;
+    });
+}
```

Comment:
This requests all items with their contents and so hides the overhead from the user. This can lead to misuse (i.e. user requests list of items and then requests some of the items separately although the items were already received).

I don't see a necessity for this function as the user can just request all items. Iterating over or extracting the IDs from the result is trivial then. Then the user also knows what they do and which overhead they create.

Also, the item IDs can already be requested via the discovery manager. This is probably the preferred option anyways. (see https://xmpp.org/extensions/xep-0060.html#entity-discoveritems)

Reply:
The content is not requested because it uses what you suggested (https://xmpp.org/extensions/xep-0060.html#entity-discoveritems). Have a look at `tst_QXmppPubSubManager::testRequestItemIds()`.

Should `requestItemIds()` use `QXmppDiscoveryManager::requestDiscoItems()`?

Reply:
Sorry, forget what I said, nvm.